src/back/wgsl/writer.rs

```diff
@@ -1034,11 +1034,11 @@ impl<W: Write> Writer<W> {
         // If the plain form of the expression is not what we need, emit the
         // operator necessary to correct that.
         let plain = plain_form_indirection(expr, module, func_ctx);
-        match (requested, plain) {
-            (Indirection::Ordinary, Indirection::Reference) => write!(self.out, "&")?,
-            (Indirection::Reference, Indirection::Ordinary) => write!(self.out, "*")?,
-            (_, _) => {}
-        }
+        let opened_paren = match (requested, plain) {
+            (Indirection::Ordinary, Indirection::Reference) => { write!(self.out, "&")?; false },
+            (Indirection::Reference, Indirection::Ordinary) => { write!(self.out, "(*")?; true },
```

Comment:
The same problem will arise with `&` operators, in theory. So both of these need to be parethesized.

Reply:
Yeah, I didn't know if `&foo.bar` was valid. I can easily fix.